code/core/src/main/scala/com.snowplowanalytics.stream/loader/transformers/JsonTransformer.scala

```diff
@@ -60,8 +60,14 @@ class JsonTransformer(
     with StdinTransformer {
 
   val log           = LoggerFactory.getLogger(getClass)
-  val dateFormatter = new SimpleDateFormat(documentIndexSuffixFormat.getOrElse(""))
-  dateFormatter.setTimeZone(TimeZone.getTimeZone("UTC"))
+
+  private val dateFormatter: Option[SimpleDateFormat] =   documentIndexSuffixFormat
+  match {
+    case Some(format) if !(format.trim.isEmpty()) => new SimpleDateFormat(format).some
+    case _ => None
+  }
+
+  private val shardingField = documentIndexSuffixField.getOrElse("derived_tstamp")
```

Comment:
This snippet doesn't seem to have been styles according to the Spotify Scala styles guide https://github.com/spotify/scio/wiki/Style-Guide